CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+# Contributing Guidelines
+
+If you are in hurry, just please scan the **Quick Start** section.
+
+## Quick Start
+
+### Ideas, bugs...
+
+- File an [issue][issues].
+- [API Blueprint][] prepares direct support for testing and scenarios. Interested?
+  Check out [apiaryio/api-blueprint#21](https://github.com/apiaryio/api-blueprint/issues/21)!
+
+### Coding
+
+- Dredd is written in [CoffeeScript][].
+- There's [ready-made virtual machine][vde] to get you started very quickly.
+
+Recommended workflow:
+
+1. Fork Dredd, create a feature branch.
+2. Write tests, use `npm run tests:bdd`.
+3. Write code.
+4. Make sure [test coverage][] didn't drop.
+5. Send a Pull Request.
+
+## Handbook for Contributers and Maintainers
+
+### Maintainers
+
+[Apiary][] is the main author and maintainer of Dredd's [upstream repository][].
+Currently responsible people are:
+
+- [@netmilk](https://github.com/netmilk) - product decisions, feature requests
+- [@honzajavorek](https://github.com/honzajavorek) - lead of development
+
+### Programming Language
+
+Dredd is written in [CoffeeScript][] and is meant to be ran on server using
+Node.js. Before publishing to npm registry, it is compiled to plain
+ES5 JavaScript code (throwaway `lib` directory).
+
+While tests are compiled on-the-fly thanks to CoffeeScript integration with
+the Mocha test framework, they actually need the code to be also pre-compiled
+every time because some integration tests use code linked from `lib`. This is
+certainly a flaw and it slows down day-to-day development, but unless we find
+out how to get rid of the `lib` dependency, it's necessary.
+
+### Linting
+
+Dredd uses [coffeelint][] to lint the CoffeeScript codebase. There is a plan
+to converge with Apiary's [CoffeeScript Style Guide][], but as most of
+the current code was written before the style guide was introduced, it's
+a long run. The effective settings are in the [coffeelint.json][] file.
+
+Linter is optional for local development to make easy prototyping and work
+with unpolished code, but it's enforced on CI level. It is recommended you
+integrate coffeelint with your favorite editor so you see violations
+immediately during coding.
+
+### Changelog
+
+[Changelog][] is maintained manually. To update it,
+
+1.  use the `npm run changelist` to generate `CHANGELOG-Generated.md` file from
+    Pull Requests merged into `master`.
+2.  Take a look on what has been generated and craft yourself a nice manual
+    update to `CHANGELOG.md`. Make it easily readable and stress achieved goals.
+
+### Documentation
+
+The main documentation is written in [Markdown][] using [MkDocs][]. Dredd uses
+[ReadTheDocs][] to build and publish the documentation:
+
+- http://dredd.readthedocs.org/ - preferred long URL
```

Comment:
Also, perhaps it would be nice to setup a CNAME using `dredd.apiary.io` or such. So that we're not tied to readthedocs and can move away easily in the future without breaking the URL.